### DISCUSSION TITLE
Actual Budget - Post Install

### DISCUSSION THREAD

**Discussion by MickLesk** [community-scripts/ProxmoxVE#807] [Guides]
![image](https://github.com/user-attachments/assets/e3a1addd-206a-4e05-a6bf-ad8975130e0b)

![image](https://github.com/user-attachments/assets/f64c909c-cd55-4c98-88e3-fa902a84bccc)

**Comment by githubaff0** [comment]
Update: The latest version of the Proxmox Helper script (as  of release 2025-02-20) enables HTTPS via a self-signed cert by default, so new installs should not run into the SharedArrayBuffer issue. 

--- Original

I just installed Actual Budget recently. I wanted use HTTPS instead of using fallback mode, so I can share some  of my notes/findings about setting up an HTTPS reverse proxy for Actual Budget.

1. I wanted to use HTTPS instead of fallback mode because of this thread: https://www.reddit.com/r/actualbudgeting/comments/1d6ldvo/how_important_is_httpssharedarraybuffer_to_not/
2. The SharedArrayBuffer issue seems to be due to different browser behavior when the browser is accessing http to localhost (no problem) vs http to non-localhost (problem) vs https to non-localhost (no problem) - https://github.com/actualbudget/actual/issues/436#issuecomment-1374924101
3. So this means you can resolve the issue with a reverse proxy terminating the HTTPS connection as an alternative to enabling HTTPS directly on your Actual Budget server.
4. The Actual software doesn't support mapping the server to a subdir path in a proxy. See https://github.com/actualbudget/actual/issues/672#issuecomment-1447079234
5. If you happen to use DuckDNS for your homelab domain/cert, note that DuckDNS will resolve subdomains of your DuckDNS domain, i.e. if you register foobar.duckdns.org, then actual.foorbar.duckdns.org will resolve as well. See https://www.reddit.com/r/homelab/comments/89li2u/comment/dwsk7qo/
6. You may run into problems if you try to register multiple DuckDNS domains (normal and subdomain or wildcard) on single cert via certbot using DNS challenge. See problem and possible solutions here: https://superuser.com/q/1839628 - I just registered a separate cert for my actual subdomain

**Reply by nickdaria** [reply]
FWIW, I simply put it behind NGINX proxy manager with SSL enabled and that made it work

**Reply by githubaff0** [reply]
> I simply put it behind NGINX proxy manager with SSL enabled and that made it work

Yes, that's basically the TL;DR for my post and that's what I did:
 * Set up an NPM proxy using a subdomain (e.g. actual.x.duckdns.org)
 * Created a new wildcard cert (e.g. *.x.duckdns.org) separate from my root cert (x.duckdns.org) instead of trying to create an 'uber' cert that covers both my root domain and wildcard subdomain.

I was trying to cite/capture all the references I found while researching the topic and going down some rabbit holes.  I should have made my original the post simpler to understand or given it a TL;DR.

**Reply by nickdaria** [reply]
> > I simply put it behind NGINX proxy manager with SSL enabled and that made it work
> 
> Yes, that's basically the TL;DR for my post and that's what I did:
> 
>     * Set up an NPM proxy using a subdomain (e.g. actual.x.duckdns.org)
> 
>     * Created a new wildcard cert (e.g. *.x.duckdns.org) separate from my root cert (x.duckdns.org) instead of trying to create an 'uber' cert that covers both my root domain and wildcard subdomain.
> 
> 
> I was trying to cite/capture all the references I found while researching the topic and going down some rabbit holes. I should have made my original the post simpler to understand or given it a TL;DR.

Oh it was very informative, I was just trying to throw a TLDR out there for people who run NGINX/NGINX PM and wanted a quick solution

**Reply by Nilsemann** [reply]
Has anybody tried this container with https://github.com/NikxDa/actual-moneymoney - I have problems to get this working, I assume it has something to do with the HTTPS connection...

**Comment by staticnom** [comment]
what is the root account username / password when going to console

**Reply by michelroegl-brunner** [reply]
There is none if you did not set one. If you get prompted the installation did not finish properly, and you need to reinstall.

**Comment by darrendavid** [comment]
Trying to update from 25.3.1 to 26.1.0 using `update` from within the LXC, and it just dumps me back out at the prompt without updating the application (I manually upgraded the LXC OS with `apt update && apt upgrade` to see if that would help, no dice.). What I see is below (verbose mode) -- am I missing something here?

```
    ___        __              __   ____            __           __ 
   /   | _____/ /___  ______ _/ /  / __ )__  ______/ /___ ____  / /_
  / /| |/ ___/ __/ / / / __ `/ /  / __  / / / / __  / __ `/ _ \/ __/
 / ___ / /__/ /_/ /_/ / /_/ / /  / /_/ / /_/ / /_/ / /_/ /  __/ /_  
/_/  |_\___/\__/\__,_/\__,_/_/  /_____/\__,_/\__,_/\__, /\___/\__/  
                                                  /____/            

Hit:1 http://deb.debian.org/debian bookworm InRelease
Hit:2 http://security.debian.org bookworm-security InRelease        
Hit:3 https://deb.nodesource.com/node_22.x nodistro InRelease       
Hit:4 http://deb.debian.org/debian bookworm-updates InRelease
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
All packages are up to date.

removed 54 packages, and changed 98 packages in 2s

16 packages are looking for funding
  run `npm fund` for details
  ✔️   Update Node.js 22
  ⏳   https://actualbudget.org/docs/backup-restore/backuproot@actualbudget:~# 
```

**Reply by potter201010** [reply]
I have this happening to me as well. Looking at the raw information it appears that it's not giving the entire message:

    msg_info "Old Installation Found, you need to migrate your data and recreate to a new container"
    msg_info "Please follow the instructions on the Actual Budget website to migrate your data"
    msg_info "https://actualbudget.org/docs/backup-restore/backup"

**Reply by MickLesk** [reply]
fixed with https://github.com/community-scripts/ProxmoxVE/pull/13860

An issue would have helped; we don't track every discussion, so it would have been fixed long ago
